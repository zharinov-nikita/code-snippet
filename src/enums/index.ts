export * from './template-name.enum'