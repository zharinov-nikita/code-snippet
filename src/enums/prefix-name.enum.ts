import type { TypeCase } from '../types'

export const enumPrefixName: TypeCase = {
  camelCase: 'prefixName',
  pascalCase: 'PrefixName',
  lowerSnakeCase: 'prefix_name',
  upperSnakeCase: 'PREFIX_NAME',
  lowerKebabCase: 'prefix-name',
  upperKebabCase: 'PREFIX-NAME',
}
