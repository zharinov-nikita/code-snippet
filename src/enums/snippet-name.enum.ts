import type { TypeCase } from '../types'

export const enumSnippetName: TypeCase = {
  camelCase: 'snippetName',
  pascalCase: 'SnippetName',
  lowerSnakeCase: 'snippet_name',
  upperSnakeCase: 'SNIPPET_NAME',
  lowerKebabCase: 'snippet-name',
  upperKebabCase: 'SNIPPET-NAME',
}
