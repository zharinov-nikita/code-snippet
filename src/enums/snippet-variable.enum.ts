import type { TypeCase } from '../types'

export const enumSnippetVariable: TypeCase = {
    camelCase: 'snippetVariable',
    pascalCase: 'SnippetVariable',
    lowerSnakeCase: 'snippet_variable',
    upperSnakeCase: 'SNIPPET_VARIABLE',
    lowerKebabCase: 'snippet-variable',
    upperKebabCase: 'SNIPPET-VARIABLE',
}
