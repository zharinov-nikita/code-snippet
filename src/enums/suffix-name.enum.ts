import type { TypeCase } from '../types'

export const enumSuffixName: TypeCase = {
  camelCase: 'suffixName',
  pascalCase: 'SuffixName',
  lowerSnakeCase: 'suffix_name',
  upperSnakeCase: 'SUFFIX_NAME',
  lowerKebabCase: 'suffix-name',
  upperKebabCase: 'SUFFIX-NAME',
}
