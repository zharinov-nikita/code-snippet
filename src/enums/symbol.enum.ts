export const enumSymbol = {
    check: 'âˆš',
    cross: 'x',
}
