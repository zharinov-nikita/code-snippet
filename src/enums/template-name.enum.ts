export enum enumTemplateName {
  camelCase = 'templateName',
  pascalCase = 'TemplateName',
  lowerSnakeCase = 'template_name',
  upperSnakeCase = 'TEMPLATE_NAME',
  lowerKebabCase = 'template-name',
  upperKebabCase = 'TEMPLATE-NAME',
}
