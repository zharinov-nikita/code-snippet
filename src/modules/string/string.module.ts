export class ModuleString {
  public toCamelCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const firstWord = array[0]
    const otherWord = array.map((item, index) => {
      if (index > 0) {
        return `${item[0].toUpperCase()}${item.substring(1)}`
      }
    })

    return `${firstWord}${otherWord.join('')}`
  }

  public toUpperSnakeCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const string = array.map((item) => item.toUpperCase())
    return string.join('_')
  }

  public toLowerSnakeCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const string = array.map((item) => item.toLowerCase())
    return string.join('_')
  }

  public toUpperKebabCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    return stringInKebabCase.toUpperCase()
  }

  public toLowerKebabCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    return stringInKebabCase.toLowerCase()
  }

  public toPascalCase(stringInKebabCase: string): string {
    if (stringInKebabCase.length === 0) return ''
    const array = stringInKebabCase.split('-')
    const firstWord = `${array[0][0].toUpperCase()}${array[0].substring(1)}`
    const otherWord = array.map((item, index) => {
      if (index > 0) {
        return `${item[0].toUpperCase()}${item.substring(1)}`
      }
    })

    return `${firstWord}${otherWord.join('')}`
  }

  private isFirstLetterUpperCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.charAt(0).toUpperCase() === stringInAnyCase.charAt(0)
  }

  private isFirstLetterLowerCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.charAt(0).toLowerCase() === stringInAnyCase.charAt(0)
  }

  private isStringContainsDash(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.search(/-/g) > 0
  }

  private isStringContainsUnderlining(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.search(/_/g) > 0
  }

  private isStringUpperCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.toUpperCase() === stringInAnyCase
  }

  private isStringLowerCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    return stringInAnyCase.toLowerCase() === stringInAnyCase
  }

  public isCamelCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isFirstLetterUpperCase(stringInAnyCase)) return false
    if (this.isStringContainsDash(stringInAnyCase)) return false
    if (this.isStringContainsUnderlining(stringInAnyCase)) return false
    if (this.isStringUpperCase(stringInAnyCase)) return false
    return true
  }

  public isPascalCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isFirstLetterLowerCase(stringInAnyCase)) return false
    if (this.isStringContainsDash(stringInAnyCase)) return false
    if (this.isStringContainsUnderlining(stringInAnyCase)) return false
    if (this.isStringUpperCase(stringInAnyCase)) return false
    return true
  }

  public isUpperSnakeCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringUpperCase(stringInAnyCase) && this.isStringContainsUnderlining(stringInAnyCase)) {
      return true
    }
    return false
  }

  public isLowerSnakeCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringLowerCase(stringInAnyCase) && this.isStringContainsUnderlining(stringInAnyCase)) {
      return true
    }
    return false
  }

  public isUpperKebabCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringUpperCase(stringInAnyCase) && this.isStringContainsDash(stringInAnyCase)) {
      return true
    }
    return false
  }

  public isLowerKebabCase(stringInAnyCase: string): boolean {
    if (stringInAnyCase.length === 0) return false
    if (this.isStringLowerCase(stringInAnyCase) && this.isStringContainsDash(stringInAnyCase)) {
      return true
    }
    return false
  }
}
