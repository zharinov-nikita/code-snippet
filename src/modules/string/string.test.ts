import { describe, expect, test } from '@jest/globals'
import { enumSnippetName } from '../../enums'
import { ModuleString } from './string.module'

describe('module/string', () => {
  const module = new ModuleString()

  const { camelCase, lowerKebabCase, lowerSnakeCase, pascalCase, upperKebabCase, upperSnakeCase } =
    enumSnippetName

  test('toCase', () => {
    expect(module.toCamelCase(lowerKebabCase)).toBe(camelCase)
    expect(module.toLowerKebabCase(lowerKebabCase)).toBe(lowerKebabCase)
    expect(module.toLowerSnakeCase(lowerKebabCase)).toBe(lowerSnakeCase)
    expect(module.toPascalCase(lowerKebabCase)).toBe(pascalCase)
    expect(module.toUpperKebabCase(lowerKebabCase)).toBe(upperKebabCase)
    expect(module.toUpperSnakeCase(lowerKebabCase)).toBe(upperSnakeCase)
  })

  test('isCamelCase', () => {
    expect(module.isCamelCase(camelCase)).toBe(true)
    expect(module.isCamelCase(lowerKebabCase)).toBe(false)
    expect(module.isCamelCase(lowerSnakeCase)).toBe(false)
    expect(module.isCamelCase(pascalCase)).toBe(false)
    expect(module.isCamelCase(upperKebabCase)).toBe(false)
    expect(module.isCamelCase(upperSnakeCase)).toBe(false)
  })

  test('isLowerKebabCase', () => {
    expect(module.isLowerKebabCase(camelCase)).toBe(false)
    expect(module.isLowerKebabCase(lowerKebabCase)).toBe(true)
    expect(module.isLowerKebabCase(lowerSnakeCase)).toBe(false)
    expect(module.isLowerKebabCase(pascalCase)).toBe(false)
    expect(module.isLowerKebabCase(upperKebabCase)).toBe(false)
    expect(module.isLowerKebabCase(upperSnakeCase)).toBe(false)
  })

  test('isLowerSnakeCase', () => {
    expect(module.isLowerSnakeCase(camelCase)).toBe(false)
    expect(module.isLowerSnakeCase(lowerKebabCase)).toBe(false)
    expect(module.isLowerSnakeCase(lowerSnakeCase)).toBe(true)
    expect(module.isLowerSnakeCase(pascalCase)).toBe(false)
    expect(module.isLowerSnakeCase(upperKebabCase)).toBe(false)
    expect(module.isLowerSnakeCase(upperSnakeCase)).toBe(false)
  })

  test('isPascalCase', () => {
    expect(module.isPascalCase(camelCase)).toBe(false)
    expect(module.isPascalCase(lowerKebabCase)).toBe(false)
    expect(module.isPascalCase(lowerSnakeCase)).toBe(false)
    expect(module.isPascalCase(pascalCase)).toBe(true)
    expect(module.isPascalCase(upperKebabCase)).toBe(false)
    expect(module.isPascalCase(upperSnakeCase)).toBe(false)
  })

  test('isUpperKebabCase', () => {
    expect(module.isUpperKebabCase(camelCase)).toBe(false)
    expect(module.isUpperKebabCase(lowerKebabCase)).toBe(false)
    expect(module.isUpperKebabCase(lowerSnakeCase)).toBe(false)
    expect(module.isUpperKebabCase(pascalCase)).toBe(false)
    expect(module.isUpperKebabCase(upperKebabCase)).toBe(true)
    expect(module.isUpperKebabCase(upperSnakeCase)).toBe(false)
  })

  test('isUpperSnakeCase', () => {
    expect(module.isUpperSnakeCase(camelCase)).toBe(false)
    expect(module.isUpperSnakeCase(lowerKebabCase)).toBe(false)
    expect(module.isUpperSnakeCase(lowerSnakeCase)).toBe(false)
    expect(module.isUpperSnakeCase(pascalCase)).toBe(false)
    expect(module.isUpperSnakeCase(upperKebabCase)).toBe(false)
    expect(module.isUpperSnakeCase(upperSnakeCase)).toBe(true)
  })
})
