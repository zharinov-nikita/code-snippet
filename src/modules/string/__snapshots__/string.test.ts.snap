// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`module/string snapshot: HELLO_WORLD 1`] = `"HELLO_WORLD"`;

exports[`module/string snapshot: HELLO-WORLD 1`] = `"HELLO-WORLD"`;

exports[`module/string snapshot: HelloWorld 1`] = `"HelloWorld"`;

exports[`module/string snapshot: hello_world 1`] = `"hello_world"`;

exports[`module/string snapshot: hello-world 1`] = `"hello-world"`;

exports[`module/string snapshot: helloWorld 1`] = `"helloWorld"`;
